tokio-test/src/task.rs

```diff
@@ -33,6 +33,7 @@ pub struct MockTask {
     waker: Arc<ThreadWaker>,
 }
 
+#[allow(clippy::mutex_atomic)]
```

Comment:
I recommended removing this lint: https://github.com/rust-lang/rust-clippy/issues/4295

Reply:
We probably should just disable this lint at the project level. This can happen in this PR or follow up work.

Reply:
Done in https://github.com/tokio-rs/tokio/pull/1335/commits/ae3bd604e97ad4776ff35d14f8e53194f074fa84.